astropix.py

```diff
@@ -0,0 +1,701 @@
+"""
+Central module of astropix. This incorporates all of the various modules from the original 
+The class methods of all the other modules are inherited here. 
+
+Author: Autumn Bauman
+"""
+# Needed modules. They all import their own suppourt libraries, 
+# and eventually there will be a list of which ones are needed to run
+from typing import Dict
+from modules.spi import Spi 
+from modules.nexysio import Nexysio
+from modules.decode import Decode
+from modules.injectionboard import Injectionboard
+from modules.voltageboard import Voltageboard
+from bitstring import BitArray
+from tqdm import tqdm
+import pandas as pd
+import regex as re
+import binascii
+import time
+
+# Logging stuff
+import logging
+from modules.setup_logger import logger
+logger = logging.getLogger(__name__)
+
+
+
+# Here are the default configuration values. 
+# This includes the DAC configurations, default registers, etc...
+
+
+
+class astropix2:
+    # First the global defaults which will be used later
+    DACS_CFG = {
+            'blres': 0,
+            'nu1': 0,
+            'vn1': 20,
+            'vnfb': 1,
+            'vnfoll': 10,
+            'nu5': 0,
+            'nu6': 0,
+            'nu7': 0,
+            'nu8': 0,
+            'vn2': 0,
+            'vnfoll2': 1,
+            'vnbias': 0,
+            'vpload': 5,
+            'nu13': 0,
+            'vncomp': 2,
+            'vpfoll': 60,
+            'nu16': 0,
+            'vprec': 30,
+            'vnrec': 30
+        }
+
+    BIAS_CFG = {
+            'DisHiDR': 0,
+            'q01': 0,
+            'qon0': 0,
+            'qon1': 1,
+            'qon2': 0,
+            'qon3': 1,
+        }
+
+
+    # Init just opens the chip and gets the handle. After this runs
+    # asic_config also needs to be called to set it up. Seperating these 
+    # allows for simpler specifying of values. 
+    def __init__(self, clock_period_ns = 10, inject:int = None):
+        """
+        Initalizes astropix object. 
+        No required arguments
+        Optional:
+        clock_period_ns:int - period of main clock in ns
+        inject:bool - if set to True will enable injection for the whole array.
+        """
+
+        # _asic_start tracks if the inital configuration has been run on the ASIC yet.
+        # By not handeling this in the init it simplifies the function, making it simpler
+        # to put in custom configurations and allows for less writing to the chip,
+        # only doing it once at init or when settings need to be changed as opposed to 
+        # each time a parameter is changed.
+        self._num_rows = 35
+        self._num_cols = 35
+
+        self._asic_start = False
+        self.nexys = Nexysio()
+        self.handle = self.nexys.autoopen()
+        self._wait_progress(2)
```

Comment:
Is this needed?

Reply:
I am not sure, it has always been there, though I changed it from 3s to 2s. We can test in the lab to see if this is actually the case or not.

Reply:
No problem, I was just wondering if you experienced any problems without this wait statement.